spec/models/manageiq/providers/redhat/infra_provider_spec.rb

```diff
@@ -18,4 +18,22 @@
     h = FactoryGirl.create(:ems_redhat, :hostname => "h")
     h.rhevm_metrics_connect_options(:hostname => "i")[:host].should == "i"
   end
+
+  it "connect Metrics" do
+    h = FactoryGirl.create(:ems_redhat, :hostname => "h")
+    expect(h).to receive(:metrics_connect)
+    h.connect(:service => "Metrics")
```

Comment:
So, as we discussed, we obviously can't let this actually go through, because it'd fail -- there isn't actually a server to connect to.

Barring the underlying library offering some sort of handy "test mode" parameter, we're not going to get out of stubbing it.

But, I think we can still improve things a bit, by focusing on the boundary from "our code" to "not our code":

```ruby
expect(OvirtMetrics).to receive(:connect).and_return(:token)
expect(h.connect(:service => "Metrics")).to eq(:token)
```

.. which does carry its own complication, in that the spec must now pre-require the library.

I do think this is a win overall, though: instead of dictating our internal class structure, we're merely asserting that eventually, we'll hit the (external, documented, and independently tested/verified) public interface of the underlying gem... and that our `connect` returns whatever that library does.